src/Paysera/WalletApi/Mapper.php

```diff
@@ -2387,4 +2401,56 @@ public function encodeTransferConfiguration(Paysera_WalletApi_Entity_TransferCon
     {
         return (new Paysera_WalletApi_Mapper_TransferConfigurationMapper())->mapFromEntity($transferConfiguration);
     }
+
+    /**
+     * @param array $data
+     * @return Paysera_WalletApi_Entity_ClientPermissionsToWallet[]
+     */
+    public function decodeClientPermissionsToWallets(array $data)
+    {
+        $result = [];
+
+        foreach ($data as $clientPermissionsToWallet) {
+            $result[] = $this->decodeClientPermissionsToWallet($clientPermissionsToWallet);
+        }
+
+        return $result;
+    }
+
+    /**
+     * Decodes client wallet permissions
+     *
+     * @param array $data
+     *
+     * @return Paysera_WalletApi_Entity_ClientPermissionsToWallet
+     */
+    public function decodeClientPermissionsToWallet(array $data)
+    {
+        return Paysera_WalletApi_Entity_ClientPermissionsToWallet::create()
+            ->setWallet($this->decodeWallet($data['wallet']))
+            ->setScopes($data['scopes'])
+        ;
+    }
+
+    /**
+     * @param Paysera_WalletApi_Entity_ClientPermissionsToWallet $walletPermissions
+     * @return array
+     * @throws Paysera_WalletApi_Exception_LogicException
+     */
+    public function encodeClientPermissionsToWallet(Paysera_WalletApi_Entity_ClientPermissionsToWallet $walletPermissions)
+    {
+        if ($walletPermissions->getWallet() === null) {
+            throw new Paysera_WalletApi_Exception_LogicException('Wallet must be provided');
+        }
+
+        if ($walletPermissions->getWallet()->getId() === null) {
+            throw new Paysera_WalletApi_Exception_LogicException('Wallet ID must be provided');
+        }
+
+        return [
+            'id' => $walletPermissions->getWallet()->getId(),
```

Comment:
wouldn't it be better to return Wallet structure of data?

Also, make sure, that on the wallet-api side the scope permissions are correctly assigned, because we show not only permissions, but also account info (account number is account info).

Reply:
In this case this structure is better than just wallet structure.

Reply:
Also I've added checking if user has permissions to read wallets list in the first step in method on wallet-api side

Reply:
I've changed naming from `id` to `wallet_id` and from `account` to `account_number`. I have no need to encode whole wallet structure to array and send it to wallet API as I resolve wallet info by wallet ID on wallet API side.